### ISSUE TITLE
[BUG] - Colours not reset after editor

### ISSUE THREAD

**Issue by Pascal-So** [veeso/termscp#94]
## Description

When I open my editor (vim) which has a background colour set, the colours are not fully reset when switching back to the filetransfer view.

See this GIF with exaggerated colours for a demonstration:

![colour](https://user-images.githubusercontent.com/18399125/149225734-35852d1d-93cc-4034-9c1b-e8640367f81a.gif)

## Steps to reproduce

1. Set a `.vimrc` with a background colour that is noticeably different from your normal terminal background. Example `.vimrc` used for the gif:

   ```vim
   highlight Normal ctermbg=darkblue
   ```
2. open termscp
3. edit a file (make sure your editor is set to vim)
4. close the editor (`:q`)
5. enjoy

## Expected behaviour

I'd expect the interface to retain the visuals from before I opened the editor.

## Environment

- OS: Arch Linux (5.15.13-arch1-1)
- Architecture: x86_64
- Rust version: 1.57.0
- termscp version: reproduced with both AUR packaged version 0.8.0 and latest main (`d1457253c480cbd852b3461389508437ee03240d`)
- Protocol used: ftp / sftp
- Remote server version and name: docker image `panubo/vsftpd:latest`

**Comment by Pascal-So** [comment]
Same applies to when the editor is opened for adding an ssh key.

**Comment by Pascal-So** [comment]
The easy fix would be to just add this snipped after [this line](https://github.com/veeso/termscp/blob/main/src/ui/activities/filetransfer/update.rs#L265):

```rust
if let Some(ctx) = &mut self.context {
    ctx.terminal.clear_screen().ok();
}
```

plus of course to all the other relevant places.

The more elegant solution might be to create a wrapper for `edit::edit_file` but there we'd probably have to pass the `TerminalBridge` into this wrapper function because not all instances of `edit::edit_file` are called from within the same activity so there is no unified way to access the `Context`.

**Comment by Pascal-So** [comment]
Do you think the same issue also applies to view file (F3|V)? When I open a `.txt` file it just opens vim in a new window for me, can you think of any way where `open::that` would end up launching a program in the same terminal window?

**Comment by veeso** [comment]
Thanks for reporting. I put this issue on the 0.8.1 backlog.

No, I don't think this should apply to `open::that` since it should always start the app in a new window.

**Comment by veeso** [comment]
I wasn't able to reproduce this issue. The clear screen is performed as soon as you leave the text editor, so it should work. Maybe there was an issue with the order this action was performed (clear screen was before entering alternate screen). But it worked for me before too, so I really don't know.

Maybe this issue happens only on your terminal? Which terminal are you using?
You could try, if you want, to build the 0.8.1 branch and see if the problem persists.

**Comment by Pascal-So** [comment]
Okay after some confusion I realized that the bug only appeared when exiting vim with `:q`, I wasn't able to reproduce it with `:wq`????? (completely agree on the "sorcery" label)

Either way, the issue seems to be fixed, and the fix stems specifically from the 4b421a4a commit, not the dependency update before that. I'm using konsole 21.12.1 on KDE. Thanks for the fix!
